packages/graphql/src/mutations/marketplace/updateListing.js

```diff
@@ -26,15 +26,6 @@ async function updateListing(_, args) {
     'ether'
   )
 
-  // Ensure that the new total units exceeds the sum of units purchased through
-  // all valid offers for this listing. This prevents unitsAvailable from going
-  // negative.
-  const newUnitsTotal = Number(ipfsData.unitsTotal) || 0
-  const listing = await marketplace.eventSource.getListing(listingId)
-  if (newUnitsTotal < listing.unitsPending) {
```

Comment:
Is not having this validation healthy?  What's the worst case of this check failing?

Reply:
Worst case the updated listing will be invalid, so we should just move the validation to the DApp as it already has access to the listing and prevent the mutation being called from there.

Reply:
Sounds good to me in short-term in order to get EventCache disabled. But long-term I think it is always preferrable to have the business logic doing these kind of checks on number of units available rather than the DApp.... mayeb add a TODO ?